### PR TITLE
Fix #4190 - GnomAD web service defaults to hg38

### LINKED ISSUE #4190: GnomAD link default changed to hg38 with the v4 release
Let's switch around to give a build only for 37. Also fixes update from v3.1 to 4.0, but note that MT is not available for 4 just yet.

### PR DESCRIPTION
**Pull request by dnil**
This PR adds a functionality or fixes a bug.

GnomAD links broken for build 37.

Screenshot 2023-11-02 at 10 23 37
leding to the now default nomad grch38:
Screenshot 2023-11-03 at 09 20 12

✅ with patch
Screenshot 2023-11-02 at 10 22 54
goes back to trusty bad old grch37.
Screenshot 2023-11-03 at 09 22 44

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. click links to gnomAD from different category variant pages

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [x] tests executed by DN

### CODE REVIEW

File: scout/server/links.py [hunk 1]
```diff
@@ -593,7 +594,11 @@ def gnomad_sv_link(variant_obj, build=37):
     else:
         url_template += f"-{variant_obj['position']}"
 
-    url_template += "?dataset=gnomad_sv_r2_1"
+    if build == 37 or variant_obj["chromosome"] in ["M", "MT"]:
+        url_template += "?dataset=gnomad_sv_r2_1"
+        return url_template
+
+    url_template += "?dataset=gnomad_sv_r4"
```
Comment (islean) [inline scout/server/links.py:601]: So just for my own curiosity: SVs in v4 are using hg38, but the MT ones still need to use the old set?

**Review by islean — APPROVED** [review]
Looks great! 🔥

File: scout/server/links.py (hunk 1 above)
Comment (dnil) [reply scout/server/links.py:601]: Yes, it's a bit of a version mess on their side. Several features are still only available in hg37, like gene mutation intolerance scores. They have a whole table of what version to use for what feature. 🙄 Management by research release cycle - they work really hard towards delivering something cool for the ASHG meeting each year.

Comment (islean) [reply scout/server/links.py:601]: I guess it is nice to use hg38 for some SVs at least!

**Pull request merged**